### PR TITLE
[L1T] Phase-2, track MET update for FW sync (new)

### PR DESCRIPTION
**Pull request by cecilecaillol**
### PR description:

This PR updates the Track MET calculation in order to match to FW and simplifies code. Changes include:

Removal of track transformation step, using instead the L1GTTInputProducer and L1TrackSelectionProducer
Use AP_FIXED data types for internal Et calculation, simplifying code and avoiding unnecessary conversions
Use TTTrack_word phi granularity extended to 2 Pi instead of internal representation
Rewrite Cordic function to use ap_fixed removing extra conversions
Cordic now runs between -pi and pi rather than 0 and 2 pi as per output specification

### PR validation:

Tested against FW, currently matches px and py calcuations as well as MET and MET phi. Differences between Emulation and Simulation are ongoing but going forwards with emulator and FW matching these differences can be investigated while also updating the FW.

Supersedes https://github.com/cms-sw/cmssw/pull/39725, which got conflicts after merging https://github.com/cms-sw/cmssw/pull/39422

Porting local https://github.com/cms-l1t-offline/cmssw/pull/1049

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by cecilecaillol** [comment]
@AdrianoDee, @srimanob Could you please check and sign? We would like this PR to be part of 12_6_0_pre4 if possible to rebase our local branch to that release.

File: L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h [hunk 1]
```diff
@@ -47,7 +47,7 @@ namespace l1tmhtemu {
     kUnassignedMSB = kUnassignedLSB + kUnassignedSize - 1
   };
 
-  const float kMaxMHT{4096};  // 4 TeV
+  const float kMaxMHT{2048};  // 4 TeV
```
Comment (AdrianoDee) [inline L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h:50]: Still valid this `4 TeV`?

File: L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h [hunk 2]
```diff
@@ -26,10 +26,10 @@ namespace l1tmhtemu {
 
   // extra room for sumPx, sumPy
   const unsigned int kEtExtra{10};
-
   const unsigned int kValidSize{1};
-  const unsigned int kMHTSize{15};     // For output Magnitude default 15
-  const unsigned int kMHTPhiSize{14};  // For output Phi default 14
+  const unsigned int kMHTSize{16};  // For output Magnitude default 15
+  const unsigned int kMHTIntSize{11};
+  const unsigned int kMHTPhiSize{13};  // For output Phi default 14
```
Comment (AdrianoDee) [inline L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h:32]: Still valid this `default 14`?

**Comment by AdrianoDee** [comment]
please test
(refresh)

File: L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h (hunk 1 above)
Comment (cecilecaillol) [reply L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h:50]: @Chriisbrown Can you answer this question?

File: L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h (hunk 2 above)
Comment (cecilecaillol) [reply L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h:32]: @Chriisbrown can you answer this question?

File: L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h (hunk 1 above)
Comment (Chriisbrown) [reply L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h:50]: The comment should be updated to 2 TeV

**Comment by cecilecaillol** [comment]
please test

**Comment by AdrianoDee** [comment]
+upgrade

**Comment by cecilecaillol** [comment]
+l1

**Comment by rappoccio** [comment]
+1


* Concerns seem to be addressed.

**Pull request merged**